presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -144,21 +144,21 @@ public PlanCostEstimate visitOutput(OutputNode node, Void context)
         public PlanCostEstimate visitTableScan(TableScanNode node, Void context)
         {
             // TODO: add network cost, based on input size in bytes? Or let connector provide this cost?
-            LocalCostEstimate localCost = LocalCostEstimate.ofCpu(getStats(node).getOutputSizeInBytes(node.getOutputVariables()));
+            LocalCostEstimate localCost = LocalCostEstimate.ofCpu(getStats(node).getOutputSizeInBytes(node));
             return costForSource(node, localCost);
         }
 
         @Override
         public PlanCostEstimate visitFilter(FilterNode node, Void context)
         {
-            LocalCostEstimate localCost = LocalCostEstimate.ofCpu(getStats(node.getSource()).getOutputSizeInBytes(node.getOutputVariables()));
+            LocalCostEstimate localCost = LocalCostEstimate.ofCpu(getStats(node).getOutputSizeInBytes(node));
```

Comment:
Lol this was a bug I guess, we should call getStats(node) instead of getStats(node.getSource())

Reply:
Can we confirm this with @rschlussel ?

Reply:
I don't think it's a bug.  The cost of a filter node should be proportional to the number of rows it processes (source stats), not the number of rows it outputs (node stats).  Could use a comment though.  It's not obvious without thinking about it.

Reply:
In that case, we should use 

`node.getSource().getOutputSizeInBytes(node.getSource())`

rather than 

`node.getSource().getOutputSizeInBytes(node)`